services/api/service.go

```diff
@@ -386,6 +405,74 @@ func (api *RelayAPI) startValidatorRegistrationDBProcessor() {
 	}
 }
 
+// simulateBlock sends a request for a block simulation to blockSimRateLimiter.
+func (api *RelayAPI) simulateBlock(opts blockSimOptions) error {
+	t := time.Now()
+	simErr := api.blockSimRateLimiter.send(opts.ctx, opts.req, opts.highPrio)
+	log := opts.log.WithFields(logrus.Fields{
+		"duration":   time.Since(t).Seconds(),
+		"numWaiting": api.blockSimRateLimiter.currentCounter(),
```

Comment:
Would it be worth logging something similar to `numWaiting` but for optimistic block processors?

Reply:
that seems like it could be valuable to provide metrics into how many of the block processors are working at a time! this is a slightly bigger change because we don't have a currentCounter like the block sim rate limiter has. let me tinker on it and get back to this thread. leaving it open for now.